### PR TITLE
make macros balancer_compatible

### PR DESCRIPTION
**Pull request by viniabussafi**
### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [x] Adding to existing spell lineage
- [ ] Bug fix

---

This PR updates the Balancer macros for easy integration with other chains/friendly forks, like Beethoven X on Fantom and Jellyverse on Sei and a few other ones potentially coming.
No logic changes are introduced and current results remain unaffected. 
The only macro not integrated is for the liquidity spell on Fantom, due to outliers in the dex.prices table. This will be addressed in a future PR.

### CODE REVIEW

**Comment by jeff-dude** [comment]
hi @viniabussafi -- thank you for continuing the balancer cleanup in spellbook!

looks like our latest move to put dex + related spells into its own subproject has introduced a few merge conflicts. since this PR has no change other than clean code, i'm assuming no rush. i will wait for you to help rebase the branch and its conflicts since i think you mentioned being on a mini vacation.

balancer had maybe the most complex lineage to move in the dex subproject, so hopefully i didn't mess all this up too much 😅  happy to work with you on getting all of it cleaned up as needed

**Comment by viniabussafi** [comment]
Hey, @jeff-dude! Thanks for your patience on this one. Have fixed the merge conflicts and change the models to use the reference the labels spell as `{{ source('labels', 'balancer_v2_pools') }}`. Should be good to go now.

> balancer had maybe the most complex lineage to move in the dex subproject, so hopefully i didn't mess all this up too much 😅 happy to work with you on getting all of it cleaned up as needed

Sorry to hear that 😅! Please let me know if there's something missing on this PR in order to match the dex sector standars

**Comment by jeff-dude** [comment]
@Hosuke if you don't mind, can you find time soon to help review this while i'm out?
after approval, you can ask @aalan3 to help with any merges required

**Comment by Hosuke** [comment]
> @Hosuke if you don't mind, can you find time soon to help review this while i'm out? after approval, you can ask @aalan3 to help with any merges required

I think I will do some testing query to check the data consistency.

File: models/beethoven_x/fantom/beethoven_x_fantom_bpt_prices.sql [hunk 1]
```diff
@@ -8,378 +8,12 @@
...
-                day,
-                contract_address,
-                median_price,
-                AVG(median_price) OVER (PARTITION BY contract_address) AS avg_median_price
-            FROM trade_price_formulation
-        ) subquery
-        WHERE median_price < avg_median_price * 2 --removes outliers
-        GROUP BY 1, 2, 3
+{{ 
+    balancer_v2_compatible_bpt_prices_macro(
```
Comment (Hosuke) [inline models/beethoven_x/fantom/beethoven_x_fantom_bpt_prices.sql:12]: Please check if the unmatched records are acceptable:
https://dune.com/queries/3869052

File: models/beethoven_x/fantom/beethoven_x_fantom_protocol_fee.sql [hunk 2]
```diff
@@ -7,148 +7,13 @@
...
-                address,
-                name,
-                pool_type,
-                ROW_NUMBER() OVER (PARTITION BY address ORDER BY MAX(updated_at) DESC) AS num
-            FROM {{ ref('labels_beethoven_x_pools_fantom') }}
-            GROUP BY 1, 2, 3) 
-        WHERE num = 1
-    ),
+{{ 
+    balancer_v2_compatible_protocol_fee_macro(
```
Comment (Hosuke) [inline models/beethoven_x/fantom/beethoven_x_fantom_protocol_fee.sql:11]: Seems matched:
https://dune.com/queries/3869067

File: models/beethoven_x/fantom/beethoven_x_fantom_bpt_supply.sql [hunk 3]
```diff
@@ -6,133 +6,12 @@
...
-        FROM joins j
-        FULL OUTER JOIN exits e ON j.block_date = e.block_date AND e.tokenIn = j.tokenOut
-    ),
-
-    calendar AS (
-        SELECT 
-            date_sequence AS day
-        FROM unnest(sequence(date('2021-04-21'), date(now()), interval '1' day)) as t(date_sequence)
+{{ 
+    balancer_v2_compatible_bpt_supply_macro(
```
Comment (Hosuke) [inline models/beethoven_x/fantom/beethoven_x_fantom_bpt_supply.sql:10]: Please check if the unmatched records are acceptable:
https://dune.com/queries/3869082

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/beethoven_x/fantom/beethoven_x_fantom_bpt_supply.sql (hunk 3 above)
Comment (viniabussafi) [reply models/beethoven_x/fantom/beethoven_x_fantom_bpt_supply.sql:10]: unmatched records are more accurate with this new macro, according to beets' subgraph

File: models/beethoven_x/fantom/beethoven_x_fantom_bpt_prices.sql (hunk 1 above)
Comment (viniabussafi) [reply models/beethoven_x/fantom/beethoven_x_fantom_bpt_prices.sql:12]: unmatched records are more accurate with this new macro, according to beets' subgraph

File: models/beethoven_x/fantom/beethoven_x_fantom_bpt_supply.sql (hunk 3 above)
Comment (Hosuke) [reply models/beethoven_x/fantom/beethoven_x_fantom_bpt_supply.sql:10]: Then I think this PR is good to go.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @viniabussafi 

This PR updates beethoven_x_fantom views' data as well.

**Comment by viniabussafi** [comment]
thanks a lot for the review, @Hosuke!

**Comment by viniabussafi** [comment]
Hey, team! Can we please merge this one anytime soon? It's holding me back on a few other things. Thank you!
@Hosuke

**Pull request merged**